pkg/test/kube/dump.go

```diff
@@ -42,6 +42,27 @@ func outputPath(workDir string, cluster resource.Cluster, pod kubeApiCore.Pod, n
 	return path.Join(dir, fmt.Sprintf("%s_%s", pod.Name, name))
 }
 
+func DumpDebug(a resource.Cluster, workDir, systemNamespace, endpoint string) {
+	errFmt := "failed dumping " + endpoint + " for " + a.Name() + ": %s: %v"
+	pods, err := a.PodsForSelector(context.TODO(), systemNamespace, "istio=pilot")
+	if err == nil && len(pods.Items) == 0 {
+		err = fmt.Errorf("0 pods found for istio: pilot in %q", systemNamespace)
+	}
+	if err == nil {
+		name := pods.Items[0].Name
+		out, _, err := a.PodExec(name, systemNamespace, "discovery", "curl 127.0.0.1:15000"+endpoint)
```

Comment:
This will break distroless, we could instead do `pilot-discovery request GET endpoint` I think